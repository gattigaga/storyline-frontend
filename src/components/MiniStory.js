import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

const StyledLink = styled(Link)`
  text-decoration: none;
`;

const Container = styled.div`
  width: 100%;
  display: flex;
  padding: 8px 0px;
`;

export const Index = styled.span`
  font-size: 32px;
  font-weight: bold;
  font-family: Roboto;
  color: #aaa;
`;

const Thumbnail = styled.img`
  width: 72px;
  height: 96px;
  object-fit: cover;
  margin: 0px 12px;
`;

const Detail = styled.div`
  flex: 1;
`;

const Title = styled.h1`
  font-family: Roboto;
  font-weight: bold;
  font-size: 16px;
  color: #333;
  margin-top: 0px;
  margin-bottom: 12px;
`;

const Author = styled.p`
  font-family: Roboto;
  font-size: 12px;
  color: #aaa;
  margin: 0px;
`;

const Date = styled.p`
  font-family: Roboto;
  font-size: 12px;
  color: #e74c3c;
  margin: 0px;
`;

const MiniStory = ({ index, image, title, href, author, date }) => {
  const indexLabel = index < 10 ? `0${index}` : `${index}`;

  return (
    <StyledLink to={href}>
      <Container>
        <Index>{indexLabel}</Index>
        <Thumbnail src={image} />
        <Detail>
          <Date>{date}</Date>
          <Title>{title}</Title>
          <Author>by {author}</Author>
        </Detail>
      </Container>
    </StyledLink>
  );
};

MiniStory.propTypes = {
  index: PropTypes.number,
  image: PropTypes.string,
  title: PropTypes.string,
  href: PropTypes.string,
  author: PropTypes.string,
  date: PropTypes.string
};

MiniStory.defaultProps = {
  index: 0,
  href: "#"
};

export default MiniStory;
