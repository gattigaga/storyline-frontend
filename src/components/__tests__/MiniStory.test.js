import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import MiniStory, { Index } from "../MiniStory";

describe("MiniStory", () => {
  const setup = propOverrides => {
    const props = {
      image: "dummy.png",
      title: "Introduction to React",
      author: "Gattigaga Hayyuta Dewa",
      date: "Jan 01, 2017",
      index: 5,
      ...propOverrides
    };

    const wrapper = shallow(<MiniStory {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with double digit index", () => {
    const { wrapper } = setup({ index: 17 });

    expect(wrapper.find(Index).prop("children")).toEqual("17");
  });
});
