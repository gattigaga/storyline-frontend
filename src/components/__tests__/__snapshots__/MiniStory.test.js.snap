// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniStory should renders default 1`] = `
<Styled(Link)
  to="#"
>
  <styled.div>
    <styled.span>
      05
    </styled.span>
    <styled.img
      src="dummy.png"
    />
    <styled.div>
      <styled.p>
        Jan 01, 2017
      </styled.p>
      <styled.h1>
        Introduction to React
      </styled.h1>
      <styled.p>
        by 
        Gattigaga Hayyuta Dewa
      </styled.p>
    </styled.div>
  </styled.div>
</Styled(Link)>
`;
